### PR TITLE
shell: fix possible deadlock in shell_fprintf

### LINKED ISSUE #27161: shell:  shell_start() and shell_stop() can cause deadlock
When using shell_start() and shell_stop() to control the start and stop of the shell when the USB is connected and disconnected (USB is used for console), the shell freezes.  It appears the freeze occurs because the underlying poll that waits for the data never receives the signal.

I previously attempted to resolve this through changes to uninit() in #24404.  It was pointed out that there are other issues with init() and uninit() and recommended I separate the two issues.

So, I removed the uninit() changes from that PR and created this issue to track the outstanding issue of starting and stopping the shell.  My use case is to disable the shell when the USB is disconnected and reenable it when the USB is reconnected.

I found a temporary work around by disabling the transmit and receive interrupts but this doesn't really "disable" and "enable" the shell ... rather, it simply avoids sending anything.

### PR DESCRIPTION
**Pull request by jakub-uC**
Disable shell print functions if the shell is not initialized.

Fixes #27161

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordisemi.no>

### CODE REVIEW

**Review by nordic-krch — APPROVED** [review]
looks good

**Comment by jakub-uC** [comment]
@Mierunski : ping

File: subsys/shell/shell_ops.h [hunk 1]
```diff
@@ -151,6 +151,16 @@ static inline void flag_last_nl_set(const struct shell *shell, uint8_t val)
 	shell->ctx->internal.flags.last_nl = val;
 }
 
+static inline bool flag_print_noinit_get(const struct shell *shell)
+{
+	return shell->ctx->internal.flags.print_noinit == 1 ? true : false;
```
Comment (Mierunski) [inline subsys/shell/shell_ops.h:156]: Comparison should be enough

File: include/shell/shell_dummy.h [hunk 2]
```diff
@@ -57,6 +57,13 @@ const struct shell *shell_backend_dummy_get_ptr(void);
 const char *shell_backend_dummy_get_output(const struct shell *shell,
 					   size_t *sizep);
 
+/**
+ * @brief Clears the output buffer in the shell backend.
+ *
+ * @param shell		Shell pointer
```
Comment (Mierunski) [inline include/shell/shell_dummy.h:63]: Too many spaces

**Review by Mierunski — CHANGES_REQUESTED** [review]

File: subsys/shell/shell_ops.h (hunk 1 above)
Comment (jakub-uC) [reply subsys/shell/shell_ops.h:156]: will fix it

File: include/shell/shell_dummy.h (hunk 2 above)
Comment (jakub-uC) [reply include/shell/shell_dummy.h]: done

**Review by Mierunski — APPROVED** [review]

**Pull request merged**